src/plopp/backends/matplotlib/figure.py

```diff
@@ -1,16 +1,24 @@
 # SPDX-License-Identifier: BSD-3-Clause
-# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
 
 from ...graphics import BaseFig
+from .utils import fig_to_bytes
 
+try:
+    from ipywidgets import VBox
+except ImportError:
+    VBox = object
```

Comment:
Is there a check elsewhere that raises a user-friendly error message? If not, this code here would lead to `AttributeError`s without further explanation if `ipywidgets` is not installed.

Reply:
No there is no check elsewhere, but I am not sure you can enable `%matplotlib widget` without having `ipympl` installed, and that itself requires `ipywidgets`?
I will double check.

Reply:
So does this code only use `VBox` when widgets are enabled? Not when a user builds custom widgets using Plopp's API?

Reply:
Result of the check: if you try to do `%matplotlib widget` in a notebook without ipywidgets installed, you get
```
RuntimeError: 'widget is not a recognised GUI loop or backend name
```

> So does this code only use VBox when widgets are enabled

Yes, the `VBox` here should only be used to create a figure when `%matplotlib widget` is enabled. If it isn't, a `StaticFigure` is created instead.

> Not when a user builds custom widgets using Plopp's API?

If a user is building custom widgets, I'm guessing they must have `ipywidgets` installed?

The try/except here is to ensure that `ipywidgets` is not a hard dependency when we import, because `InteractiveFigure` inherits from `VBox`.
Maybe there is a better way to do this?

Reply:
I'm sure there are more complicated ways to do it and have a custom error message. But I think it's fine. I was just worried about people getting bad error messages from Plopp. But if they have to manually import ipywidgets anyway when building custom widgets, that's fine.